_posts/2022-03-16-async-sink-base.md

```diff
@@ -0,0 +1,160 @@
+---
+layout: post
+title: "The Generic Asynchronous Base Sink"
+date: 2022-04-05 16:00:00
+authors:
+- CrynetLogistics:
+  name: "Zichen Liu"
+excerpt: An overview of the new AsyncBaseSink and how to use it for building your own concrete sink
+---
+
+Flink sinks share a lot of similar behavior. All sinks batch records according to user defined buffering hints, sign requests, write them to the destination, retry unsuccessful or throttled requests, and participate in checkpointing.
+
+This is why for [Flink 1.15](https://cwiki.apache.org/confluence/display/FLINK/FLIP-171%3A+Async+Sink) we have decided to create the `AsyncSinkBase`, an abstract sink with a number of common functionalities extracted. Adding support for a new destination now only requires a lightweight shim that implements the specific interfaces of the destination using a client that supports async requests. 
+
+This common abstraction will reduce the effort required to maintain individual sinks that extend from this abstract sink, with bugfixes and improvements to the sink core benefiting all implementations that extend it. The design of AsyncSyncBase focuses on extensibility and a broad support of destinations. The core of the sink is kept generic and free of any connector-specific dependencies.
+
+The sink base is designed to participate in checkpointing to provide at-least-once semantics and can work directly with destinations that provide a client that supports asynchronous requests. Alternatively, concrete sink implementers may manage their own thread pool with a synchronous client.
+
+{% toc %}
```

Comment:
```suggestion
In this post, we will go over some of the details of the AsyncSinkBase so that you can start using it to build your own concrete sink. 

{% toc %}
```

Reply:
I recommend a minor change to "In this post, we will go over -some of -the details of the AsyncSinkBase so that you can start using it to build your own concrete sink. "